sklearn/utils/estimator_checks.py

```diff
@@ -1037,7 +1038,60 @@ def check_estimator_sparse_data(name, estimator_orig):
             err_msg = (
                 f"Estimator {name} doesn't seem to support {matrix_format} "
                 "matrix, and is not failing gracefully, e.g. by using "
-                "check_array(X, accept_large_sparse=False)"
+                "check_array(X, accept_large_sparse=False)."
+            )
+        else:
+            err_msg = (
+                f"Estimator {name} doesn't seem to fail gracefully on sparse "
+                "data: error message should state explicitly that sparse "
+                "input is not supported if this is not the case."
+            )
+        with raises(
```

Comment:
Could we share most of the code between `check_estimator_sparse_matrix` and `check_estimator_sparse_array` by creating a function `_check_estimator_sparse_container` that take a sparse container (array or matrix). Then we could call this private function in the `check_estimator_sparse_matrix` and `check_estimator_sparse_array`.

Reply:
I did something similar, I hope it's okay.